### ISSUE TITLE
Remove MAMP recommendation from docs

### ISSUE THREAD

**Issue by unclecheese** [silverstripe/silverstripe-framework#7772]
MAMP is no longer a viable option for running SS4 in a dev environment due to the lack of `php-intl` extension. We should remove this from [the documentation](https://docs.silverstripe.org/en/4/getting_started/installation/mac_osx/) and recommend something else. Vagrant? Homestead?

**Comment by kinglozzer** [comment]
I’ve done SS4 development on MAMP Pro before and intl was available. According to `phpinfo();` there was a `--enable-intl` flag passed when building PHP - seems to be the same for the pre-installed versions, and other versions installed via the UI.

Not sure if/how this might differ from the free version, though...

**Comment by unclecheese** [comment]
Interesting. Every time someone in the community is trying to do this, I google it, and it's just one [overly complex solution](http://aerendir.me/2015/07/17/how-to-install-php-intl-module-in-mamp/)  after another, and TBH I can't imagine recommending something like that. I think it will make our software look more complicated than it is.

**Comment by lerni** [comment]
I can confirm latest MAMP Pro 4.2.1 with php 5.6.31, 7.0.22 & 7.1.8 all are compiled with --enable-intl set. I guess it's available with recent versions but couldn't find a reference.

**Comment by sminnee** [comment]
Given @lerni's comment is this ticket still valid?

**Comment by aledbrown** [comment]
I asked MAMP Support about it and I got the following response which I haven't attempted yet. Hopefully it's helpful to others using MAMP Pro 4 and SS4.

---
The command line PHP uses a different php.ini file.

If you are using PHP 7.2 command line in MAMP PRO, then it will be using this php.ini

/Applications/MAMP/bin/php/php7.2.2/conf

—

Try the following,

1. Rename this file to php7.2.2.ini_bak
2 Run MAMP PRO, this will create a php.ini file in /Library/Application Support/appsolute/MAMP PRO/conf/php.7.2.2.ini file. Take this php7.2.2.ini file and copy it to /Applications/MAMP/bin/php/php7.2.2/conf

Now try running your command line PHP.

**Comment by kinglozzer** [comment]
Bear in mind you may also need to create/update your symlinks in `/usr/local/bin`. I believe OS X ships with a pre-installed copy of PHP, and I don’t think MAMP automatically overrides that when installing different PHP versions.

You can test that by running `/Applications/MAMP/bin/php/php7.2.2/bin/php -m` and checking if `intl` is in that list. If it is, then intl is set up as expected but Composer and PHP CLI are using a different PHP installation.

**Comment by aledbrown** [comment]
I think you're right I'm using two PHPs at the moment!

In MAMP Pro 4 on the Languages > PHP screen you can tick "Make this version available on the command line" and it'll use the version selected as the CLI PHP, overriding whatever you've got set on macOS. Currently I have my upgraded macOS PHP as the CLI because it's working but I can flick the switch in MAMP to have that one, which I'll try setting up again with php-intl when I have time. Wasted too much already.

I like using MAMP Pro because it boxes off the Dev environment from the System so when the time comes to change Macs or for disaster recovery/backup it's a lot easier to get back set-up. It's worked for me for many years reliably and MAMP Pro itself handles updates quite well.

**Comment by trevorpan** [comment]
I think I may have a similar post here:
https://forum.silverstripe.org/t/proper-setup-of-bash-profile-for-mamp/764

Is the intl.so issue mamp and not .bash_profile? I tried uncle cheese's link above and it gave me a similar error.

Is MAMP the best local dev software to develop SS on?

**Comment by aledbrown** [comment]
For today I've set my CLI PHP to this:
PHP_VERSION=`ls /Applications/MAMP/bin/php/php7.1.12/bin/php`
and when you run php -m to see the modules list, intl is there so I'm hoping that's all there is to it.
Unless Composer also needs to be made aware of that PHP.

**Comment by dhensby** [comment]
> Is MAMP the best local dev software to develop SS on?

No - hence this issue saying we should remove MAMP from recommendations. Use Vagrant to build a native dev environment that reflects your production servers

**Comment by aledbrown** [comment]
I disagree that MAMP is necessarily a bad one. IMO It certainly should not be removed from the SS setup notes because it's so popular. The guys at MAMP have great support and are receptive to making changes to support apps and I'm sure they'd make SS4 setups easier if someone from SS got in touch?

Lots of people need to use Shared Hosting providers as well which simply don't have access to any of the dev commands your local machine might have and still want to use a high quality CMS like SilverStripe. MAMP is perfect for this.

Vagrant or the like is great if you're working on a high budget site with a VPS you can re-create the condition of locally.

IMHO the very simplicity of setting up a great site quickly with SilverStripe is being overshadowed by the complexity of getting a dev environment setup. It used to be easier in the SS2/3 days.

**Comment by dhensby** [comment]
We are removing the recommendation of MAMP because we can't provide support for it when people raise tickets. MAMP allows far too many variations and by saying "hey, install with MAMP like this..." when something goes wrong people come to us to debug their MAMP install. That's not something we can do effectively.

People are free to use MAMP, but they should be aware of it's "gotchas" and limitations **which apply to all PHP projects, not just SS**

Whilst a Vagrant box (and even a MAMP install) gives you access to commands that you wouldn't have with shared hosting, they aren't commands you have to use... Once you get your software dependencies installed (correct PHP version, etc) then don't use any server side commands you can't in your shared hosting.

In my mind, Vagrant is not exclusively for high budget work, it's just a simple way to set up a dev environment that isn't polluted by any other project. It's a sandbox to work in so you know that the software you needed for project x doesn't conflict with your new project y. There is no project too small to be run on a Vagrant box.

There should be absolutely nothing special or custom required to get SilverStripe running on a *AMP set-up. The fact you say that we should contact the MAMP maintainers to make it play nicely with SS only goes to show that it is not an easy solution.

For example, give me any linux box and I can install the required PHP deps, mysql and apache and I'll have SS running for you in 5 mins; it sounds like MAMP is much more complicated.

>IMHO the very simplicity of setting up a great site quickly with SilverStripe is being overshadowed by the complexity of getting a dev environment setup. It used to be easier in the SS2/3 days.

If you have any particular pain points regarding the set up of hosting (that aren't specific to a 3rd party piece of software you use to run it), we'd love to hear it so we can address those problems.

It really should be a simple as installing php and the required modules, apache and mysql, then `composer create-project` and you're away.

**Comment by aledbrown** [comment]
I am going to try the Vegrant Workflow after looking it up. It does sound like a big leap forward. I'll go look up the SS install documentation for using it. Thanks for the info.

As for pain points it's not necessarily new projects with SS4 that are the problem. They're actually easy to setup. One line and then you can begin coding. That's actually just perfect and fine.

I'm probably projecting my frustration with the SS3 to SS4 upgrade process more. As a long time SS user I am struggling with it. Some part might have been dev box setup as well. I think I'm going to have to do a from scratch project first then I might understand the upgrade more. Lots of lost time so far.

**Comment by dhensby** [comment]
@aledbrown have you tried using the upgrade tool? https://github.com/silverstripe/silverstripe-upgrader

We've had really good feedback on it helping internal and external teams upgrade their projects. We also want feedback on the upgrade process and how we can make that better too; a lot of time has gone into the upgrader and if there's anything it's not doing for you, open an issue and we'll take it on board.

> I am going to try the Vegrant Workflow after looking it up. It does sound like a big leap forward. I'll go look up the SS install documentation for using it. Thanks for the info.

With Vagrant I'd recommend finding a SS box to use rather than provisioning one yourself (unless you're confident doing that). I maintain this one: https://github.com/betterbrief/vagrant-skeleton but there are others (here's one: https://www.twistedbytes.nl/en/blog/php-vagrant-box/)

**Comment by aledbrown** [comment]
Yeah I was using the upgrader tool and tying myself in knots with different upgrade to SS4 documentation pages. Some were referencing the old mysite directory and then others with the new app directory structure. I got to displaying my theme in SS4 and the CMS admin but the database content would not load into either after one upgrader command did something and lost the link between classes, I never recovered from there on. I need to start that upgrade from the top. I ended up having to continue with SS3 and getting the job done. I was also battling the php intl issue while doing the SS4 upgrade so it might have been a contributing factor.

Thanks for the Vagrant SS Box links, that looks really great. I'm going to follow that through and use it on an SS project.

**Comment by trevorpan** [comment]
Ok great. Certainly not on your guys level of understanding but I'll give vagrant a shot. Just had a lot of issues with MAMP and the pathing using the .bash_profile. Much appreciated!

**Comment by dhensby** [comment]
@aledbrown sorry to hear the pain you were having. Perhaps that was an older version of the upgrader; it's come on a lot recently.

The docs consistency is a problem, and wherever you see an issue like you've described it would be hugely appreciated if you could submit a fix to them, though I appreciate that can be difficult when you're short on time.

**Comment by aledbrown** [comment]
It was just week or two ago with SS 4.1.2 jumping form SS 3.5 ish (been able to run it on SS 3.7.1 no problem - job completed). I will have to get time to try SS4 again but time is critical for all projects and I lost more than half a day to it. I was really enjoying the new file path structures and the beefed up security that will bring. I hope I just made a mistake somewhere.

I couldn't find a fan site for SS4. There used to be loads back in the SS2 days. SS3 had less but wasn't a difficult upgrade. Missing the old community.

**Comment by jonom** [comment]
Vagrant is great, and I'd agree it should be the default recommendation. I use it for a few sites but I do notice a performance hit compared to running a local server which can be frustrating. I use a homebrew AMP stack for most local development, and this tutorial has been very helpful over the years, particularly after MacOS updates as the guide is routinely updated: https://getgrav.org/blog/macos-sierra-apache-multiple-php-versions Perhaps a link to that could be included for any homebrew fans.

**Comment by aledbrown** [comment]
I can confirm that the recent MAMP Pro 5 paid update fixes all the SS4 issues I had. I am now up and running with an updated SS3 website to SS4.2.1 and update-code and composer are both working completely as expected under PHP 7.1.19 and MySQL 5.7.

I've also repaired all my "failed" SS4 upgrades that I'd started and abandoned and picked up loads of tips and tricks for the upgrades. It's still a lot of work but at least I've finally got myself some working instances to compare and work from. Hopefully I can progress from here now.
